### DISCUSSION TITLE
Problem using Package org.apache.commons.io

### DISCUSSION THREAD

**Discussion by cranicaudal** [nextgenhealthcare/connect#6139] [Q&A] [answered]
Hi everybody,
hope someone can help me.
I've a Problem when using Package org.apache.commons.io
I want to move a FIle if its Size ist biger than 150 MB. So i write a Script that looks at the Size and than try to move it to another Directory. But i cn't get it to Run.
I alway get the Error:  Can't find method org.apache.commons.io.FileUtils.moveFileToDirectory
What i tried is: I downloaded the binaries from Org Apache and put it in the Custom Library of Mirth Directory

Thank`s for help
Here ist the code:

```
importPackage(org.apache.commons.io);
var sourcepath = 'srcDir';
var destDir = 'dest';
var dir = new java.io.File(sourcepath).listFiles();
	try{
		var file = new Packages.java.io.File(sourcepath);
		if(file.exists()) {
			var list = file.list();
				for(i=0;i<list.length;i++) {
					var Filetomove = new Packages.java.io.File(file,list[i]);
					var Filesizemove = Filetomove.length();			
					if(Filesizemove > 150000000){				
					FileUtils.moveFileToDirectory(Filetomove, destDir);
						}
					}
				}
		}
	catch (e){
		logger.error (("Fehler im") + " " + channelName + " " + "Fehler: " + e.toString());
		}

```

**Comment by JoeFox82** [comment]
Hi,
please take a look at https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html#moveFileToDirectory(java.io.File,java.io.File,boolean)
The function _moveFileToDirectory_ has three parameters.

If it is still not working please try using _importPackage(org.apache.commons.io.**FileUtils**);_ in the first row of your script

Kinds regards

**Comment by cranicaudal** [comment]
Hi JoeFox82,
thanks for your answer.

I put in the Third Parameter like this: FileUtils.moveFileToDirectory(Filetomove, destDir, true);

The error is just the same:
 Can't find method org.apache.commons.io.FileUtils.moveFileToDirectory(java.io.File,string,boolean)

I also put  importPackage(org.apache.commons.io.FileUtils); in the first row .
I get another Error:
Function importPackage must be called with a package; had "[JavaClass org.apache.commons.io.FileUtils]" instead.

**Reply by cranicaudal** [reply]
I now have tried out several other methods the only method that works was: forceDelete
All other methods are not found.

**Comment by rjkroll** [comment]
This works for me in a transformer step without having to declare any packages: 

```
var defaultSrcDirfile = '//server1/share/PDF/filename.pdf';
var defaultDestDirfile = '//server2/share/PDF/filename.pdf';
var src =  new java.io.File(defaultSrcDirfile);
var dest =  new java.io.File(defaultDestDirfile);
org.apache.commons.io.FileUtils.moveFile(src,dest);
```

**Reply by cranicaudal** [reply, marked as answer]
Hi rjkroll,
thanks a lot this works even with the Method moveToDirectory

```
var sourcepath = 'srcDir';
var destDir = 'dest';
var dir = new java.io.File(sourcepath).listFiles();

	try{
		var file = new Packages.java.io.File(sourcepath);
		if(file.exists()) {
			var list = file.list();
				for(i=0;i<list.length;i++) {
					var Filetomove = new Packages.java.io.File(file,list[i]);
					var Filesizemove = Filetomove.length();			
					if(Filesizemove > 10){
						var defaultSrcDirfile = Filetomove;
						var defaultDestDirfile = destDir;
						var src = new java.io.File(defaultSrcDirfile);
						var dest =  new java.io.File(defaultDestDirfile); 
						org.apache.commons.io.FileUtils.moveToDirectory(src,dest,true);
												}
					}
				}
		}
	catch (e){
		logger.error (("Fehler im") + " " + channelName + " " + "Fehler: " + e.toString());
		}
```
